### PR TITLE
Issue 90 gbert mlflow integration

### LINKED ISSUE #90: Adapt gbert template - cycle 3
* Integrate MLflow
* Integrate extended validation sets

### PR DESCRIPTION
**Pull request by dominikmn**
* Fix: _Augmented_ val set is taken for validation.
* Fix: _Best_ epoch is logged to MLflow (instead of _last_ epoch).
* Enhancement: F-beta score is computed in addition to F1 score.

Closes #90

### CODE REVIEW

File: modeling/gbert_classifier.py [hunk 1]
```diff
@@ -224,52 +260,112 @@ def make_model(label):
...
+    mlflow_logger.add_param("grid_search_params", str(constant_params)[:249]) 
+    mlflow_logger.add_param("lr", learning_rate)
+    mlflow_logger.add_param('pos_weight', positive_class_weight)
+
+    mlflow_logger.add_param("saved_model", f"model_gbertbase_{label}_{t}")
+    mlflow_logger.add_param("label", data.current_label)
+    mlflow_logger.add_param("balance_method", data.balance_method)
+    if data.balance_method:
+        mlflow_logger.add_param("sampling_strategy", data.sampling_strategy)
+    mlflow_logger.add_model(None)
```
Comment (lima-tango) [inline modeling/gbert_classifier.py:340]: FYI: As you never set a model, you should not need to do this. It is by default set to None and therefore, not logging.
But it might be better to leave it in here, to have it explicit.

File: modeling/gbert_classifier.py [hunk 2]
```diff
@@ -224,52 +260,112 @@ def make_model(label):
...
+    constant_params = {
+                    'epochs': EPOCHS,
+                    'batch_size': BATCH_SIZE,
+                    'max_len': MAX_LEN,
+                }
+    mlflow_logger.add_tag("cycle4", True)
+    mlflow_logger.add_param("normalization", 'norm')
+    mlflow_logger.add_param("vectorizer", 'deepset/gbert-base')
+    mlflow_logger.add_param("model", "deepset/gbert-base")
+    # I'm re-using the grid_search_params field in order to no open too many mlflow columns
```
Comment (lima-tango) [inline modeling/gbert_classifier.py:330]: typo: no -> not

File: modeling/gbert_classifier.py [hunk 3]
```diff
@@ -184,38 +179,79 @@ def eval_model(model, data_loader, loss_fn, device, n_examples):
                 input_ids=input_ids,
                 attention_mask=attention_mask
                 )
-            preds = torch.round(outputs)
+            preds = torch.sigmoid(outputs)
```
Comment (lima-tango) [inline modeling/gbert_classifier.py:182]: why did you move the sigmoid into the `eval_model`?

**Review by lima-tango — APPROVED** [review]
looks good! Thanks for the cleaning that you have already done!

File: modeling/gbert_classifier.py (hunk 3 above)
Comment (dominikmn) [reply modeling/gbert_classifier.py:183]: Question has been answered in the team call meanwhile. :heavy_check_mark:

For reference: Because of using the different loss function `BCEWithLogitsLoss`

File: modeling/gbert_classifier.py (hunk 2 above)
Comment (dominikmn) [reply modeling/gbert_classifier.py]: Typo resolved in b7d6ab255560e5719b55d6496850204160730750
